src/parser.ts

```diff
@@ -0,0 +1,272 @@
+    .trim()
+    .substr(1)
+    .trim();
+
+  const addedInfo: Partial<VariableDeclaration> = {};
+  if (line.startsWith('required:')) {
+    const val = textToBoolean(removePrefix('required:', line));
+    addedInfo.required = val;
+  } else if (line.startsWith('format:')) {
+    const val = textToFormat(removePrefix('format:', line));
+    addedInfo.format = val;
+  } else if (line.startsWith('description:')) {
+    const val = trim(removePrefix('description:', line));
+    addedInfo.description = val;
+  } else if (line.startsWith('link:')) {
+    const val = trim(removePrefix('link:', line));
+    addedInfo.link = val;
+  } else if (line.startsWith('default:')) {
+    const val = trim(removePrefix('default:', line));
+    addedInfo.default = val;
+    addedInfo.hasExplicitDescription = true;
+  } else if (line.startsWith('configurable:')) {
+    const val = textToBoolean(removePrefix('configurable:', line));
+    addedInfo.configurable = val;
+  } else {
+    if (!currentDeclaration.hasExplicitDescription) {
+      addedInfo.description = currentDeclaration.description
+        ? `${currentDeclaration.description} - ${line}`
+        : line;
+    }
+  }
+
+  return { ...currentDeclaration, ...addedInfo };
+}
+
+/**
+ * Parses a variable declaration line and merges the key and default value with the
+ * current variable declaration meta info. Throws an error for invalid keys
+ * @param currentDeclaration the variable declaration parsed so far
+ * @param line the variable declaration line to parse
+ */
+export function parseVariableDeclarationLine(
+  currentDeclaration: VariableDeclaration,
+  line: string
+): VariableDeclaration {
+  const [key, value] = line.split('=').map(trim);
+
+  if (key.match(INVALID_DECLARATION_CHARACTERS)) {
+    throw new Error(
+      `Key of variable declaration can only contain the following characters. Received: "${key}"`
+    );
+  }
+  const defaultValue = value.length === 0 ? currentDeclaration.default : value;
+
+  return { ...currentDeclaration, key, default: defaultValue };
+}
+
+/**
+ * Parses a .env file and extracts meta information according to the
+ * [schema specified in the docs](../docs/SCHEMA.md). Returns the parsed
+ * variable information and a mustache-style template to fill in data.
+ *
+ * @param envFileContent .env in string representation
+ */
+export function parse(envFileContent: string): ParseResult {
+  const variables: VariableDeclaration[] = [];
+  const outputTemplateLines: string[] = [];
+
+  const lines = envFileToLines(envFileContent);
+
+  let currentVariable = DEFAULT_ENTRY;
+  for (const line of lines) {
+    if (line.startsWith('#')) {
+      // line is a comment
+      currentVariable = parseCommentLine(currentVariable, line);
+    } else if (line.includes('=')) {
+      // line is a variable declaration
+      const fullVariableData = parseVariableDeclarationLine(
+        currentVariable,
+        line
+      );
+      delete fullVariableData.hasExplicitDescription;
```

Comment:
why is the deletion operation being performed?

Reply:
I had to maintain a state during the parsing to make sure that it's properly handled if a description is explict (using `description:`) vs implicit (using just comments). I'm deleting it since it's not part of the spec and I don't see it being valuable for anyone outside of the parser.